src/libslic3r/PerimeterGenerator.cpp

```diff
@@ -1325,6 +1356,15 @@ static ExtrusionEntityCollection traverse_extrusions(const PerimeterGenerator& p
                     multi_path.paths.emplace_back(std::move(*it_path));
                 }
 
+				//This is for staggered layers. 
+				//All odd perimeters are staggerd up by half the layer height                
+                if (extrusion->inset_idx % 2 == 1 && perimeter_generator.config->staggered_perimeters) {
```

Comment:
Same here.
Change to `if (perimeter_generator.config->staggered_perimeters && extrusion->inset_idx % 2 == 1)` for better performance